docs/design.rst

```diff
@@ -23,7 +28,33 @@ Kafka is used as the Schema Registry storage backend. The special Kafka topic ``
 Single Master Architecture
 ~~~~~~~~~~~~~~~~~~~~~~~~~~
 The Schema Registry is designed to work as a distributed service using single master architecture. In this configuration, at most one Schema Registry instance is master at any given moment (ignoring pathological 'zombie masters'). Only the master is capable of publishing writes to the underlying Kafka log, but all nodes are capable of directly serving read requests. Slave nodes serve registration requests indirectly by simply forwarding them to the current master, and returning the response supplied by the master.
+Prior to Schema Registry version 4.0, master election was always coordinated through Zookeeper.
+Master election can now optionally happen via Kafka Coordinator protocol as well.
+
+Zookeeper Master Election
++++++++++++++++++++++++++
+
+.. figure:: schema-registry-design.png
+   :align: center
+
+   Zookeeper based Schema Registry
+
+Zookeeper maser election is chosen when Zookeeper URL is specified in the Schema Registry config
+``<kafkastore.connection.url>``.
+The current master is maintained as data in the ephemeral node on the``/<schema.registry.zk.namespace>/schema_registry_master`` path in ZooKeeper. Schema Registry nodes listen to data change and deletion events on this path, and shutdown or failure of the master process triggers each node with ``master.eligibility=true`` to participate in a new round of election. Master election is a simple 'first writer wins' policy: the first node to successfully write its own data to ``/<schema.registry.zk.namespace>/schema_registry_master`` is the new master.
+
+Kafka Coordinator Master Election
++++++++++++++++++++++++++++++++++
+
+.. figure:: schema-registry-design-kafka.png
+   :align: center
+
+   Kafka based Schema Registry
 
-The current master is maintained as data in the ephemeral node on the ``/<schema.registry.zk.namespace>/schema_registry_master`` path in ZooKeeper. Schema Registry nodes listen to data change and deletion events on this path, and shutdown or failure of the master process triggers each node with ``master.eligibility=true`` to participate in a new round of election. Master election is a simple 'first writer wins' policy: the first node to successfully write its own data to ``/<schema.registry.zk.namespace>/schema_registry_master`` is the new master.
+Kafka coordinator based master elction is done when Schema Registry configs has no Zookeeper
```

Comment:
"when Schema Registry configs has no Zookeeper" sounds like you're referring to configs across multiple instances. should probably just say "when the ``<kafkastore.connection.url>`` configuration is not specified and ..."